### PR TITLE
fix(iec): move datasource[iec_flavors,iec_images,iec_keypair] into services and fix codecheck errors

### PR DESCRIPTION
**Pull request by zhangting951007**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
move datasource[iec_flavors,iec_images,iec_keypair] into services and fix codecheck errors

**Special notes for your reviewer**:

**Release note**:

```release-note
./scripts/codecheck.sh ./huaweicloud/services/iec

==> Checking for running environment...

==> Applying patch...

==> Checking for code complexity...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        3       303       32         0      271         25            27.69
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~/iec/data_source_huaweicloud_iec_images.go       112       11         0      101         10             9.90
~iec/data_source_huaweicloud_iec_flavors.go       127       12         0      115         10             8.70
~iec/data_source_huaweicloud_iec_keypair.go        64        9         0       55          5             9.09
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     3       303       32         0      271         25            27.69
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 7584 bytes, 0.008 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP10 most complex functions:
7 iec dataSourceIecImagesRead huaweicloud/services/iec/data_source_huaweicloud_iec_images.go:66:1
7 iec dataSourceIecFlavorsRead huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go:79:1
4 iec dataSourceIECKeypairRead huaweicloud/services/iec/data_source_huaweicloud_iec_keypair.go:39:1
1 iec DataSourceIECKeypair huaweicloud/services/iec/data_source_huaweicloud_iec_keypair.go:18:1
1 iec DataSourceIecImages huaweicloud/services/iec/data_source_huaweicloud_iec_images.go:17:1
1 iec DataSourceIecFlavors huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go:17:1
Average: 3.5

==> Checking for golangci-lint...

==> Checking for Nolint directives...

==> Checking for TF features in iec...

==> Checking for misspell in iec...

==> Checking for code complexity in ./huaweicloud/services/acceptance/iec...
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Language                              Files     Lines   Blanks  Comments     Code Complexity Complexity/Lines
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Go                                        3       176       24         0      152          6            10.95
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
~ata_source_huaweicloud_iec_flavors_test.go        81       11         0       70          3             4.29
~data_source_huaweicloud_iec_images_test.go        53        8         0       45          3             6.67
~ata_source_huaweicloud_iec_keypair_test.go        42        5         0       37          0             0.00
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Total                                     3       176       24         0      152          6            10.95
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Processed 4901 bytes, 0.005 megabytes (SI)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────

the TOP5 most complex functions:
3 iec testAccCheckIECImagesDataSourceID huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_images_test.go:33:1
3 iec testAccCheckIECFlavorsDataSourceID huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_flavors_test.go:52:1
1 iec testAccDataSourceIECKeypair_basic huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_keypair_test.go:32:1
1 iec TestAccIECKeypairDataSource_basic huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_keypair_test.go:13:1
1 iec testAccIECImagesConfig huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_images_test.go:47:1
Average: 1.4

==> Checking for golangci-lint in ./huaweicloud/services/acceptance/iec...

==> Checking for Nolint directives in ./huaweicloud/services/acceptance/iec...

==> Cleanup patch...

Check Completed!

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec' TESTARGS='-run TestAccIECFlavorsDataSource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test /home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec -v -run TestAccIECFlavorsDataSource -timeout 360m -parallel 4
=== RUN   TestAccIECFlavorsDataSource_basic
=== PAUSE TestAccIECFlavorsDataSource_basic
=== RUN   TestAccIECFlavorsDataSource_FilterName
=== PAUSE TestAccIECFlavorsDataSource_FilterName
=== CONT  TestAccIECFlavorsDataSource_basic
=== CONT  TestAccIECFlavorsDataSource_FilterName
--- PASS: TestAccIECFlavorsDataSource_FilterName (10.63s)
--- PASS: TestAccIECFlavorsDataSource_basic (10.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       10.998s
make testacc TEST='/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec' TESTARGS='-run TestAccIECImagesDataSource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test /home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec -v -run TestAccIECImagesDataSource -timeout 360m -parallel 4
=== RUN   TestAccIECImagesDataSource_basic
=== PAUSE TestAccIECImagesDataSource_basic
=== CONT  TestAccIECImagesDataSource_basic
--- PASS: TestAccIECImagesDataSource_basic (9.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       9.424s
make testacc TEST='/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec' TESTARGS='-run TestAccIECKeypairDataSource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test /home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec -v -run TestAccIECKeypairDataSource -timeout 360m -parallel 4
=== RUN   TestAccIECKeypairDataSource_basic
=== PAUSE TestAccIECKeypairDataSource_basic
=== CONT  TestAccIECKeypairDataSource_basic
--- PASS: TestAccIECKeypairDataSource_basic (19.72s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       19.762s

```

### CODE REVIEW

File: huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go [hunk 1]
```diff
@@ -73,12 +76,12 @@ func dataSourceIecFlavors() *schema.Resource {
 	}
 }
 
-func dataSourceIecFlavorsV1Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
+func dataSourceIecFlavorsRead(d *schema.ResourceData, meta interface{}) error {
+	cfg := meta.(*config.Config)
 
-	iecClient, err := config.IECV1Client(GetRegion(d, config))
+	iecClient, err := cfg.IECV1Client(common.GetRegion(d, cfg))
```
Comment (deer-hang) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go]:
```suggestion
	iecClient, err := cfg.IECV1Client(cfg.GetRegion(d))
```
Please check the other parts

File: huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go [hunk 2]
```diff
@@ -73,12 +76,12 @@ func dataSourceIecFlavors() *schema.Resource {
...
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud IEC client: %s", err)
+		return fmt.Errorf("error creating iec client: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go]:
```suggestion
		return fmt.Errorf("error creating IEC client: %s", err)
```
`IEC` is an abbreviation and needs to be capitalized, please check the other parts

File: huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_flavors_test.go [hunk 3]
```diff
@@ -53,11 +53,11 @@ func testAccCheckIECFlavorsDataSourceID(n string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[n]
 		if !ok {
-			return fmtp.Errorf("Root module has no resource called %s", n)
+			return fmt.Errorf("root module has no resource called %s", n)
 		}
 
 		if rs.Primary.ID == "" {
-			return fmtp.Errorf("IEC flavors data source ID not set")
+			return fmt.Errorf("iec flavors data source ID not set")
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/iec/data_source_huaweicloud_iec_flavors_test.go]: IEC is the cloud service name, should be capitalized. please check them all.
```suggestion
			return fmt.Errorf("IEC flavors data source ID not set")
```

File: huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go [hunk 4]
```diff
@@ -90,18 +91,18 @@ func dataSourceIecFlavorsV1Read(d *schema.ResourceData, meta interface{}) error
...
 	if err != nil {
-		return fmtp.Errorf("Unable to extract iec flavors: %s", err)
+		return fmt.Errorf("unable to extract IEC flavors: %s", err)
 	}
 	total := len(allFlavors.Flavors)
 	if total < 1 {
-		return fmtp.Errorf("Your query returned no results of huaweicloud_iec_flavors. " +
-			"Please change your search criteria and try again.")
+		return fmt.Errorf("your query returned no results of iec_flavors," +
+			"please change your search criteria and try again")
```
Comment (ShiChangkuo) [inline huaweicloud/services/iec/data_source_huaweicloud_iec_flavors.go]: please add an space at the end of `,` and check it all.
```suggestion
		return fmt.Errorf("your query returned no results of iec_flavors, " +
			"please change your search criteria and try again")
```

**Pull request merged**